src/python/dxpy/nextflow/nextflow_templates.py

```diff
@@ -66,33 +66,21 @@ def get_nextflow_src(custom_inputs=None, profile=None, resources_dir=None):
     with open(os.path.join(str(get_template_dir()), get_source_file_name()), 'r') as f:
         src = f.read()
 
-    required_runtime_params = ""
-    generate_runtime_config= ""
+    applet_runtime_params = ""
     for i in custom_inputs:
         value = "${%s}" % (i['name'])
         if i.get("class") == "file":
             value = "dx://${DX_WORKSPACE_ID}:/$(echo ${%s} | jq .[$dnanexus_link] -r | xargs -I {} dx describe {} --json | jq -r .name)" % i['name']
-        # optional inputs will be added to custom runtime config file
-        if "Nextflow pipeline optional" in i.get("help", ""):
-            if i.get("class") not in ("int","float","boolean"):
-                value = '\\"' + value + '\\"'
-            generate_runtime_config = generate_runtime_config + '''
-            if [ -n "$%s" ]; then
-                echo params.%s=%s >> nxf_runtime.config
-            fi    
-            '''% (i['name'], i['name'], value)
-        # required inputs need to be added as runtime pipeline params
-        else:
-            # required_inputs variable is initialized in the nextflow.sh script template
-            required_runtime_params = required_runtime_params + '''
-            if [ -n "${}" ]; then
-                required_inputs="${{required_inputs}} --{}={}" 
-            fi
-            '''.format(i['name'], i['name'], value)
+
+        # applet_runtime_inputs variable is initialized in the nextflow.sh script template
+        applet_runtime_params = applet_runtime_params + '''
+        if [ -n "${}" ]; then
+            applet_runtime_inputs="${{applet_runtime_inputs}} --{}={}" 
```

Comment:
What was the reason of having the required parameters separately from the optional ones before? I remember we talked about it a lot due to some edge cases

Reply:
It was in  APPS-1390 where we tried to handle this case where we try to add this default value parsed from nextflow.config
```
ribo_database_manifest     = "${projectDir}/assets/rrna-db-defaults.txt"
```
Adding this explicitly would not work
```
nextflow run ...
'--ribo_database_manifest=${projectDir}/assets/rrna-db-defaults.txt' 
```
because the projectDir is not resolved when resolving this explicit param. Only when we added it to a runtime config it would work since projectDir is a configuration implicit variable
https://www.nextflow.io/docs/latest/script.html?highlight=projectdir#configuration-implicit-variables

However 
1. passing params via -c is not recommended in DSL 2 
2. this option is no longer needed in the nextflow run command, as we already make all schema inputs as OPTIONAL applet input. Thus when one schema option is not specified by user, its default value would be honored.
@mhrvol

Reply:
Sounds good. Thanks for clarification!